### PR TITLE
allow Sensitive type for content param

### LINKED ISSUE #114: Allow Sensitive type for systemd::dropin_file::content
Since content this is just passed through to a file resource allow to pass Sensitive values also

### PR DESCRIPTION
**Pull request by TheMeier**
fixes #114

### CODE REVIEW

File: manifests/dropin_file.pp [hunk 1]
```diff
@@ -47,18 +47,18 @@
...
-  String                            $owner         = 'root',
-  String                            $group         = 'root',
-  String                            $mode          = '0444',
-  Boolean                           $show_diff     = true,
-  Enum['lazy', 'eager']             $daemon_reload = 'lazy',
+  Systemd::Unit                               $unit,
+  Systemd::Dropin                             $filename      = $name,
+  Enum['present', 'absent', 'file']           $ensure        = 'present',
+  Stdlib::Absolutepath                        $path          = '/etc/systemd/system',
+  Optional[Variant[String,Sensitive[String]]] $content       = undef,
```
Comment (raphink) [inline manifests/dropin_file.pp]: It might be clearer to make that a `Variant[String,Sensitive[String],Undef]`

Comment (TheMeier) [reply manifests/dropin_file.pp]: @raphink hm IMHO `Optional` is the more idomatic way. Also grepping over a couple of modules i have checked out here from various sources it seems `Optional[Variant]` is what is usually used:

```
$ find . -name "*.pp" |xargs grep -E "Variant.*Undef" | wc -l
40
$ find . -name "*.pp" |xargs grep -E "Optional\[Variant" | wc -l
492
```

Comment (bastelfreak) [reply manifests/dropin_file.pp]: I go with @TheMeier here. The default should be `undef` and that requires the type to be `Optional`

Comment (raphink) [reply manifests/dropin_file.pp]: @bastelfreak 	technically it doesn't require to use `Optional`, as a variant allowing `Undef` works as well.

**Comment by vchepkov** [comment]
I think it should be `Optional[Variant[String,Sensitive]]`
Sensitive is type on it's own

**Pull request merged**